unit/goto-programs/goto_program_validate.cpp

```diff
@@ -0,0 +1,359 @@
+/*******************************************************************\
+ Module: Unit tests for goto program validation
+
+ Author: Diffblue Ltd.
+
+ Date: Oct 2018
+
+\*******************************************************************/
+
+/// \file
+/// Unit tests for goto program validation
+
+#include <testing-utils/use_catch.h>
+
+#include <testing-utils/message.h>
+
+#include <goto-programs/goto_convert_functions.h>
+#include <goto-programs/validate_goto_model.h>
+
+#include <util/arith_tools.h>
+#include <util/c_types.h>
+#include <util/std_code.h>
+#include <util/validate.h>
+
+namespace
+{
+bool test_for_fail(
+  goto_modelt &goto_model,
+  goto_model_validation_optionst &validation_options)
+{
+  REQUIRE_THROWS_AS(
+    validate_goto_model(
+      goto_model.goto_functions,
+      validation_modet::EXCEPTION,
+      validation_options),
+    incorrect_goto_program_exceptiont);
+  return true;
+}
+
+bool test_for_pass(
```

Comment:
This function is a bit weird, the way you're using it you're basically having nested requires. Either make this a `void` function (and just call it instead of wrapping it in `REQUIRE`) or just inline the check within the tests themselves (may even be clearer that way, even if it's a bit more typing).

Reply:
Hmm. Rationale here was:
a) act as an interface (robust to future changes), and also demonstrate commonality 
(so people don't check in multiple lines multiple times for possible minor differences)

b) removing boiler plate : 11 vs 66 lines. Makes the core of the tests more legible.

c)  about the nested require - the intention was to preserve the familiar structure at the call site - intention was to reduce the "this is a bit weird" factor. 

Happy to remove the nested require - although I think it could be said that it serves a purpose here.

Reply:
Gone for the less surprising option - inlined all the checks.